Game/Src/UI/HealthUI.gd

```diff
@@ -1,20 +1,25 @@
 extends Control
 
 onready var stats = $StatsGen
-
 onready var green =$Green
 onready var red = $Red
+onready var size = red.rect_size.x
+export var max_health = 100
+
+export(PackedScene) var statsScript
 
-var max_damage
+var health_unit
 
 func _ready():
-	max_damage = green.rect_scale.x
-	
-func set_damage(value):
-	print('health changed by ',value)
-	GlobalVariables.damage = clamp(value, 0, max_damage)
-	if red != null:
-		red.rect_scale.x = value
-		
+	health_unit = size / max_health
+	print('Health unit ',size,' ',health_unit,' ',max_health)
+	pass
 
+func change_red(value:float)->void:
+	print('changing red ',value,' ',value*health_unit)
+	green.rect_size.x = value*health_unit
+	print(value)
 
+func _on_PlayerStats_health_changed(value:float):
+	change_red(value)
```

Comment:
change_green if the previous comment made sense
And useless pass

Reply:
Let's check the code: Indeed, it should be change green, I didn't even pay attention :P